src/rgw/rgw_sal.h

```diff
@@ -504,23 +514,23 @@ class User {
     /** Get the display name for this User */
     virtual std::string& get_display_name() { return info.display_name; }
     /** Get the tenant name for this User */
-    const std::string& get_tenant() { return info.user_id.tenant; }
+    virtual const std::string& get_tenant() { return info.user_id.tenant; }
```

Comment:
@dang now that we can have filter stores that wrap other stores, isn't it time to get rid of the member variables in the sal base classes?

if we have a `TracerUser` wrapping a `RadosUser`, we shouldn't store a `RGWUserInfo info` in both - that's bound to cause confusion and bugs

now that these getter functions are virtual, we can make them pure virtual in `rgw::sal::User` and move the member variables into the backends like `RadosUser`/`DBUser`/`MotrUser`. same with the other sal base classes

Reply:
agree, have already requested this

Reply:
Already working on this, mostly complete.  Was waiting for this PR, so as not to break all Daniel's work.